File: Vaidehi Gupta/ARRAY/majority element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
       map<int,int>mp;
       for(int i=0;i<nums.size();i++)
       {
        mp[nums[i]]++;
       } 
       int m=(nums.size()/2)+1;int n;
       for (auto i = mp.begin(); i != mp.end(); i++) 
       {
        if(i->second>=m)
        n= i->first;
        
       }

    return n;
    }
};

time complexity:O(n)
space complexity:O(n)

File: Tarun/Array/Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        Map<Integer, Integer> map = new HashMap<>();
        for (int x : nums) {
            map.put(x, map.getOrDefault(x, 0) + 1);
        }
        for (int i = 0; i < n; i++) {
            if (map.get(nums[i]) > n / 2) {
                return nums[i];
            }
        }       
        return 0; 
    }
}

Space Complexity: O(n)
Time Complexity: O(n)
